mne_bids/write.py

```diff
@@ -1493,7 +1493,8 @@ def mark_bad_channels(ch_names, descriptions=None, *, bids_path,
     bads = _get_bads_from_tsv_data(tsv_data)
     raw.info['bads'] = bads
     # XXX (How) will this handle split files?
-    raw.save(raw.filenames[0], overwrite=True, verbose=False)
+    if raw.filenames[0].endswith('.fif'):
+        raw.save(raw.filenames[0], overwrite=True, verbose=False)
```

Comment:
indeed it does not seem to be the right fix.

I would write to fif if we have MEG data / are dealing if meg modality.
what if you have CTF, KIT meg data?

Reply:
> I would write to fif if we have MEG data / are dealing if meg modality.

We always want to overwrite the input data, so using the same filename is fine. The tricky part is choosing the correct writer; e.g. BV data we'd need to manually pass to `pybv` for writing instead of just calling `Raw.save`